src/lib/registration/storage.rb

```diff
@@ -131,14 +132,18 @@ def export
       end
 
       def import(settings)
+        Yast.import "AutoinstConfig"
+
         reset
 
+        product = Yast::AutoinstConfig.selected_product.short_name
+
         @do_registration = settings.fetch("do_registration", false)
         @reg_server = settings["reg_server"] || ""
         @slp_discovery = settings.fetch("slp_discovery", false)
         @reg_server_cert = settings["reg_server_cert"] || ""
         @email = settings["email"] || ""
-        @reg_code = settings["reg_code"] || ""
+        @reg_code = settings["reg_code"] || RegCodes.instance.reg_codes[product] || ""
```

Comment:
I think it is wrong to put such logic to this layer. I think it should go higher. @lslezak can you advise please?

Reply:
Probably the [import in the scc_auto](https://github.com/yast/yast-registration/blob/master/src/clients/scc_auto.rb#L125) client.

And it should be probably skipped in the AY config mode (as mentioned above).